docs/Controllers.md

```diff
@@ -0,0 +1,99 @@
+
+Controllers SHOULD observe and interpret all of the TXT records returned with the DNS Service Discovery responses according to the requirements for Query API Clients specified in the [API Paths section of the APIs document](APIs.md#api-paths) in this specification.
+
+## Query API
+
+The Controller MUST be capable of using the Registry's IS-04 Query API to discover any registered resource, including Node, Device,  Source, Flow, Sender, and Receiver,
+as described in the [APIs section](README.md#apis) of this specification.
+
+The Controller MUST use the Registry's IS-04 Query API either via the HTTP API or by requesting WebSocket subscriptions.
+
+When using the Query API, query filters SHOULD be used (and advanced query language where available) to cut down on the volume of resources returned to the Controller, as specified in the [APIs: Query Parameters](APIs%20-%20Query%20Parameters.md) document.
+
+Controllers SHOULD adhere to the version downgrade requirements for Query API Clients specified in the [Requirements for Query API Clients section of the Upgrade Path document](Upgrade%20Path.md#requirements-for-query-api-clients) in this specification.
+
+## Pagination
+
+In large systems, API resources rely upon pagination in order to return high volumes of data.
+This can be both hard to keep track of for a client, and require a large number of requests in order to scan the entire data set.
+
+For this reason it is RECOMMENDED to use the RESTful QUERY API for debug and development purposes only.
+
+If using the RESTful Query API rather than WebSockets, Pagination requirements MUST be implemented as specified in the [Pagination section of the APIs: Query Parameters document](APIs%20-%20Query%20Parameters.md#pagination) in this specification.
+	
+## WebSockets & Subscriptions	
+
+Where a WebSocket or other subscription based mechanism is provided for Controller usage, it is strongly RECOMMENDED that Controllers make use of this and do not use the API resources directly.
+
+If a WebSocket connection fails, then an attempt to reconnect to the WebSocket SHOULD NOT be attempted. Instead, a new subscription SHOULD be created with this API or a different one if required. If all available APIs return errors, an exponential backoff algorithm SHOULD be used when retrying until a success code is returned.
+
+## Dynamic Update of Resources
+The Controller MUST be capable of using the Registry's IS-04 Query API to discover and dynamically update the state of any registered resource, including Node, Device,  Source, Flow, Sender, and Receiver.
+
+* The Controller MUST indicate available Senders to the user.
+* The Controller MUST reflect changes in presence/absence of Senders to the user after a maximum of 30 seconds.
+* The Controller MUST indicate available Receivers to the user which have an IS-05 Connection API.
+  However, the Controller can choose not to display discovered Receivers without an IS-05 Connection API.
```

Comment:
I'm asking myself why the language is different for Senders and Receivers?
Do we REQUIRE Controllers to be able to connect (IS-05 supporting) Receivers to IS-04 only Senders?